backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java

```diff
@@ -629,8 +629,9 @@ public Dataset updateDatasetName(String datasetId, String datasetName)
               .setWorkspaceTypeValue(datasetObj.getWorkspace_type())
               .build();
       // Check entity already exists
-      checkDatasetAlreadyExist(dataset);
+      checkDatasetAlreadyExist(session, dataset);
```

Comment:
I think it's fine to keep this within the transaction. It will help ensure consistency. We should just remove rpc from within the transactions.

Reply:
`checkDatasetAlreadyExist(dataset);` is calling from transaction scope so it will lead the transaction under transaction so I have make it with a single transaction.

Reply:
Please add a TODO to remove the excess of transactions. Not a big deal here since it's not a very frequent path.

Reply:
Reverted changes (`remove the excess of transactions`)

Reply:
along with TODO , please file a jira referring the TODO

Reply:
Jira Ticket: https://vertaai.atlassian.net/browse/VR-4405